### ISSUE TITLE
releated K8S windows - pods are crashing due to CNI panic #3389

### ISSUE THREAD

**Issue by digeler** [Azure/acs-engine#3447]
so now we dont have the panic any more ,but still pods are crashing ,looking at the azure-vnet.log we see the following /:
E0708 14:11:53.813677    4960 cni.go:259] Error adding network: netplugin failed but error parsing its diagnostic message "{\n    \"cniVersion\": \"0.3.0\",\n    \"interfaces\": [\n        {\n            \"name\": \"eth0\"\n        }\n    ],\n    \"dns\": {}\n}{\n    \"code\": 100,\n    \"msg\": \"A Compute System with the specified identifier does not exist.\"\n}": invalid character '{' after top-level value
E0708 14:11:53.813677    4960 cni_windows.go:49] error while adding to cni network: netplugin failed but error parsing its diagnostic message "{\n    \"cniVersion\": \"0.3.0\",\n    \"interfaces\": [\n        {\n            \"name\": \"eth0\"\n        }\n    ],\n    \"dns\": {}\n}{\n    \"code\": 100,\n    \"msg\": \"A Compute System with the specified identifier does not exist.\"\n}": invalid character '{' after top-level value
W0708 14:11:53.813677    4960 docker_sandbox.go:353] failed to read pod IP from plugin/docker: NetworkPlugin cni failed on the status hook for pod "prod-wsc-cliprouploader-57878b4f9b-dkz8j_wsc": netplugin failed but error parsing its diagnostic message "{\n    \"cniVersion\": \"0.3.0\",\n    \"interfaces\": [\n        {\n            \"name\": \"eth0\"\n        }\n    ],\n    \"dns\": {}\n}{\n    \"code\": 100,\n    \"msg\": \"A Compute System with the specified identifier does not exist.\"\n}": invalid character '{' after top-level value
2018/07/08 14:11:54 Going to send Telemetry report to hostnetagent http://169.254.169.254/machine/plugins?comp=netagent&type=cnireport
2018/07/08 14:11:54 "Start Flag false CniSucceeded false Name CNI Version v1.0.7 ErrorMessage A Compute System with the specified identifier does not exist. vnet [] 
				Context AzureCNI SubContext "
2018/07/08 14:11:54 OrchestratorDetails &{  kubectl command failed due to exit status 1}
2018/07/08 14:11:54 OSDetails &{windows    }
2018/07/08 14:11:54 SystemDetails &{0 0 0 0 0 0 }
2018/07/08 14:11:54 InterfaceDetails &{Primary 10.240.0.0/12 10.240.0.127 00:0d:3a:37:ea:e6 vEthernet (Ethernet 3) 30 0 }
2018/07/08 14:11:54 BridgeDetails <nil>
2018/07/08 14:11:54 Send telemetry success 200
2018/07/08 14:11:54 SetReportState succeeded

looking at the code /:
//looks like the issue starts here /:
W0708 14:11:53.813677    4960 docker_sandbox.go:353] failed to read pod IP from plugin/docker: 

	ip, err := ds.getIPFromPlugin(sandbox)
	if err == nil {
		return ip
	}

// getIPFromPlugin interrogates the network plugin for an IP.
func (ds *dockerService) getIPFromPlugin(sandbox *dockertypes.ContainerJSON) (string, error) {
	metadata, err := parseSandboxName(sandbox.Name)
	if err != nil {
		return "", err
	}
	msg := fmt.Sprintf("Couldn't find network status for %s/%s through plugin", metadata.Namespace, metadata.Name)
	cID := kubecontainer.BuildContainerID(runtimeName, sandbox.ID)
	networkStatus, err := ds.network.GetPodNetworkStatus(metadata.Namespace, metadata.Name, cID)
	if err != nil {
		return "", err
	}
	if networkStatus == nil {
		return "", fmt.Errorf("%v: invalid network status for", msg)
	}
	return networkStatus.IP.String(), nil
}

// getIP returns the ip given the output of `docker inspect` on a pod sandbox,
// first interrogating any registered plugins, then simply trusting the ip
// in the sandbox itself. We look for an ipv4 address before ipv6.
func (ds *dockerService) getIP(podSandboxID string, sandbox *dockertypes.ContainerJSON) string {
	if sandbox.NetworkSettings == nil {
		return ""
	}
	if networkNamespaceMode(sandbox) == runtimeapi.NamespaceMode_NODE {
		// For sandboxes using host network, the shim is not responsible for
		// reporting the IP.
		return ""
	}

	// Don't bother getting IP if the pod is known and networking isn't ready
	ready, ok := ds.getNetworkReady(podSandboxID)
	if ok && !ready {
		return ""
	}

	ip, err := ds.getIPFromPlugin(sandbox)
	if err == nil {
		return ip
	}

	// TODO: trusting the docker ip is not a great idea. However docker uses
	// eth0 by default and so does CNI, so if we find a docker IP here, we
	// conclude that the plugin must have failed setup, or forgotten its ip.
	// This is not a sensible assumption for plugins across the board, but if
	// a plugin doesn't want this behavior, it can throw an error.
	if sandbox.NetworkSettings.IPAddress != "" {
		return sandbox.NetworkSettings.IPAddress
	}
	if sandbox.NetworkSettings.GlobalIPv6Address != "" {
		return sandbox.NetworkSettings.GlobalIPv6Address
	}

	// If all else fails, warn but don't return an error, as pod status
	// should generally not return anything except fatal errors
	// FIXME: handle network errors by restarting the pod somehow?
	glog.Warnin**gf("failed to read pod IP from plugin/docker: %v", err)**
	return ""
}

so we are restarting the pod but this is not a good idea.
seems like docker is not getting ip from the plugin.
am not able to locate this source : 708 14:11:53.813677    4960 cni_windows.go:49] error while adding to cni network: netplugin failed but error parsing its diagnostic message.

your help is highly appreciated.

**Comment by PatrickLang** [comment]
Can you share more details like the JSON passed into acs-engine generate (with passwords/keys removed)? I haven't been able to reproduce this problem.

I'm using acs-engine built from master, and haven't seen any problems. Here's what the typical flow looks like for me in `c:\k\azure-vnet.log`

```
2018/07/09 18:17:48 SendReport failed due to [Azure CNI] HTTP Post returned statuscode 500
2018/07/09 18:17:51 [cni-net] Plugin azure-vnet version v1.0.7.
2018/07/09 18:17:51 [cni-net] Running on windows
2018/07/09 18:17:51 [net] Network interface: {Index:12 MTU:1500 Name:vEthernet (Ethernet 2) HardwareAddr:00:0d:3a:06:9e:a1 Flags:up|broadcast|multicast} with IP addresses: [fe80::d58a:5fd4:c65:e2bb/64 10.240.0.97/12]
2018/07/09 18:17:51 [net] Network interface: {Index:1 MTU:-1 Name:Loopback Pseudo-Interface 1 HardwareAddr: Flags:up|loopback|multicast} with IP addresses: [::1/128 127.0.0.1/8]
2018/07/09 18:17:51 [net] Network interface: {Index:9 MTU:1500 Name:vEthernet (nat) HardwareAddr:00:15:5d:52:68:9b Flags:up|broadcast|multicast} with IP addresses: [fe80::955c:e30a:ccf7:2b6b/64 172.25.48.1/20]
2018/07/09 18:17:51 [net] reboot time 0001-01-01 00:00:00 +0000 UTC store mod time 2018-07-09 18:17:46.8161681 +0000 UTC
2018/07/09 18:17:51 [net] Restored state, &{Version:v1.0.7 TimeStamp:2018-07-09 18:17:46.8021947 +0000 UTC ExternalInterfaces:map[vEthernet (Ethernet 2):0xc04215a0c0] store:0xc042132420 Mutex:{state:0 sema:0}}
2018/07/09 18:17:51 [cni-net] Plugin started.
2018/07/09 18:17:51 [cni-net] Processing ADD command with args {ContainerID:14f26dd86c33d85aa4a77d58487d826ca64cdf0f7b38c466852dc710f60e117e Netns:none IfName:eth0 Args:IgnoreUnknown=1;K8S_POD_NAMESPACE=default;K8S_POD_NAME=iis-1803-5ffd8b84d6-h4gvl;K8S_POD_INFRA_CONTAINER_ID=14f26dd86c33d85aa4a77d58487d826ca64cdf0f7b38c466852dc710f60e117e Path:/opt/azure-vnet/bin;c:\k\azurecni\bin}.
2018/07/09 18:17:51 [cni-net] Read network configuration &{CNIVersion:0.3.0 Name:azure Type:azure-vnet Mode:bridge Master: Bridge:azure0 LogLevel: LogTarget: Ipam:{Type:azure-vnet-ipam Environment: AddrSpace: Subnet: Address: QueryInterval:} DNS:{Nameservers:[10.0.0.10 168.63.129.16] Domain: Search:[svc.cluster.local] Options:[]} AdditionalArgs:[{Name:EndpointPolicy Value:[123 34 69 120 99 101 112 116 105 111 110 76 105 115 116 34 58 91 34 49 48 46 50 52 48 46 48 46 48 47 49 50 34 44 34 49 48 46 50 52 48 46 48 46 48 47 49 50 34 93 44 34 84 121 112 101 34 58 34 79 117 116 66 111 117 110 100 78 65 84 34 125]} {Name:EndpointPolicy Value:[123 34 68 101 115 116 105 110 97 116 105 111 110 80 114 101 102 105 120 34 58 34 49 48 46 48 46 48 46 48 47 49 54 34 44 34 78 101 101 100 69 110 99 97 112 34 58 116 114 117 101 44 34 84 121 112 101 34 58 34 82 79 85 84 69 34 125]}]}.
2018/07/09 18:17:51 [net] Found existing endpoint through hcsshim: &{Id:c73c9f41-64dc-4a38-80a2-f250d7e34c59 Name:14f26dd8-eth0 VirtualNetwork:83f8ea46-0286-4d1b-9059-0c390923e35c VirtualNetworkName:azure Policies:[[123 32 34 69 120 99 101 112 116 105 111 110 76 105 115 116 34 32 58 32 91 32 34 49 48 46 50 52 48 46 48 46 48 47 49 50 34 44 32 34 49 48 46 50 52 48 46 48 46 48 47 49 50 34 32 93 44 32 34 84 121 112 101 34 32 58 32 34 79 117 116 66 111 117 110 100 78 65 84 34 32 125] [123 32 34 68 101 115 116 105 110 97 116 105 111 110 80 114 101 102 105 120 34 32 58 32 34 49 48 46 48 46 48 46 48 92 47 49 54 34 44 32 34 78 101 101 100 69 110 99 97 112 34 32 58 32 116 114 117 101 44 32 34 84 121 112 101 34 32 58 32 34 82 79 85 84 69 34 32 125] [123 32 34 84 121 112 101 34 32 58 32 34 76 50 68 114 105 118 101 114 34 32 125]] MacAddress:00-15-5D-BF-74-C6 IPAddress:10.240.0.113 DNSSuffix:default.svc.cluster.local DNSServerList:10.0.0.10,168.63.129.16 GatewayAddress:10.240.0.1 EnableInternalDNS:false DisableICC:false PrefixLength:12 IsRemoteEndpoint:false Namespace:<nil>}
2018/07/09 18:17:51 [net] Attaching ep c73c9f41-64dc-4a38-80a2-f250d7e34c59 to container 14f26dd86c33d85aa4a77d58487d826ca64cdf0f7b38c466852dc710f60e117e
2018/07/09 18:17:51 [cni-net] ADD command completed with result:Interfaces:[{Name:eth0 Mac: Sandbox:}], IP:[{Version:4 Interface:<nil> Address:{IP:10.240.0.113 Mask:fff00000} Gateway:10.240.0.1}], Routes:[{Dst:{IP:0.0.0.0 Mask:00000000} GW:10.240.0.1}], DNS:{Nameservers:[10.0.0.10 168.63.129.16] Domain: Search:[] Options:[]} err:<nil>.
2018/07/09 18:17:51 [cni-net] Plugin stopped.
2018/07/09 18:17:53 SendReport failed due to [Azure CNI] HTTP Post returned statuscode 500
```

**Comment by PatrickLang** [comment]
You mention they're crashing - can you share the `kubectl describe` output showing what the sequence was? Do they start and fail after some time, or never start at all?

**Comment by digeler** [comment]
I redeployed no issues till now.
Am still looking.
Thanks

**Comment by PatrickLang** [comment]
Alright - I tried to scale a web server up to 40 instances on a single host. Things started failing with about 25 pods running. It seems problems start once IPs are exhausted.

azure-vnet-ipam.log
```
2018/07/11 00:13:11 [cni-ipam] Plugin started.
2018/07/11 00:13:11 [cni-ipam] Processing ADD command with args {ContainerID:08f9d3612335450c76c7647de4731131a8552d6664ba23401b4f5dbf2ee0e767 Netns:none IfName:eth0 Args:IgnoreUnknown=1;K8S_POD_NAMESPACE=default;K8S_POD_NAME=iis-1803-5ffd8b84d6-84pb9;K8S_POD_INFRA_CONTAINER_ID=08f9d3612335450c76c7647de4731131a8552d6664ba23401b4f5dbf2ee0e767 Path:/opt/azure-vnet/bin;c:\k\azurecni\bin}.
2018/07/11 00:13:11 [cni-ipam] Read network configuration &{CNIVersion:0.3.0 Name:azure Type:azure-vnet Mode:bridge Master: Bridge:azure0 LogLevel: LogTarget: Ipam:{Type:azure-vnet-ipam Environment: AddrSpace: Subnet:10.240.0.0/12 Address: QueryInterval:} DNS:{Nameservers:[10.0.0.10 168.63.129.16] Domain: Search:[svc.cluster.local] Options:[]} AdditionalArgs:[{Name:EndpointPolicy Value:[123 34 69 120 99 101 112 116 105 111 110 76 105 115 116 34 58 91 34 49 48 46 50 52 48 46 48 46 48 47 49 50 34 44 34 49 48 46 50 52 48 46 48 46 48 47 49 50 34 93 44 34 84 121 112 101 34 58 34 79 117 116 66 111 117 110 100 78 65 84 34 125]} {Name:EndpointPolicy Value:[123 34 68 101 115 116 105 110 97 116 105 111 110 80 114 101 102 105 120 34 58 34 49 48 46 48 46 48 46 48 47 49 54 34 44 34 78 101 101 100 69 110 99 97 112 34 58 116 114 117 101 44 34 84 121 112 101 34 58 34 82 79 85 84 69 34 125]}]}.
2018/07/11 00:13:11 [ipam] Starting source azure.
2018/07/11 00:13:11 [ipam] Refreshing address source.
2018/07/11 00:13:11 [ipam] Save succeeded.
2018/07/11 00:13:11 [ipam] Requesting address with address: options:map[].
2018/07/11 00:13:11 [ipam] Address request completed with address:<nil> err:<nil>.
2018/07/11 00:13:11 [azure-vnet-ipam] Failed to allocate address: No available addresses.
2018/07/11 00:13:11 [cni-ipam] ADD command completed with result:<nil> err:Failed to allocate address: No available addresses.
2018/07/11 00:13:11 [cni-ipam] Plugin stopped.

```

azure-vnet.log
```
2018/07/11 00:13:11 [net] Network interface: {Index:12 MTU:1500 Name:vEthernet (Ethernet 2) HardwareAddr:00:0d:3a:06:9e:a1 Flags:up|broadcast|multicast} with IP addresses: [fe80::d58a:5fd4:c65:e2bb/64 10.240.0.97/12]
2018/07/11 00:13:11 [net] Network interface: {Index:1 MTU:-1 Name:Loopback Pseudo-Interface 1 HardwareAddr: Flags:up|loopback|multicast} with IP addresses: [::1/128 127.0.0.1/8]
2018/07/11 00:13:11 [net] Network interface: {Index:9 MTU:1500 Name:vEthernet (nat) HardwareAddr:00:15:5d:52:68:9b Flags:up|broadcast|multicast} with IP addresses: [fe80::955c:e30a:ccf7:2b6b/64 172.25.48.1/20]
2018/07/11 00:13:11 [net] reboot time 0001-01-01 00:00:00 +0000 UTC store mod time 2018-07-11 00:13:09.2801396 +0000 UTC
2018/07/11 00:13:11 [net] Restored state, &{Version:v1.0.7 TimeStamp:2018-07-11 00:13:09.0370934 +0000 UTC ExternalInterfaces:map[vEthernet (Ethernet 2):0xc0420f8240] store:0xc04210a8a0 Mutex:{state:0 sema:0}}
2018/07/11 00:13:11 [cni-net] Plugin started.
2018/07/11 00:13:11 [cni-net] Processing ADD command with args {ContainerID:08f9d3612335450c76c7647de4731131a8552d6664ba23401b4f5dbf2ee0e767 Netns:none IfName:eth0 Args:IgnoreUnknown=1;K8S_POD_NAMESPACE=default;K8S_POD_NAME=iis-1803-5ffd8b84d6-84pb9;K8S_POD_INFRA_CONTAINER_ID=08f9d3612335450c76c7647de4731131a8552d6664ba23401b4f5dbf2ee0e767 Path:/opt/azure-vnet/bin;c:\k\azurecni\bin}.
2018/07/11 00:13:11 [cni-net] Read network configuration &{CNIVersion:0.3.0 Name:azure Type:azure-vnet Mode:bridge Master: Bridge:azure0 LogLevel: LogTarget: Ipam:{Type:azure-vnet-ipam Environment: AddrSpace: Subnet: Address: QueryInterval:} DNS:{Nameservers:[10.0.0.10 168.63.129.16] Domain: Search:[svc.cluster.local] Options:[]} AdditionalArgs:[{Name:EndpointPolicy Value:[123 34 69 120 99 101 112 116 105 111 110 76 105 115 116 34 58 91 34 49 48 46 50 52 48 46 48 46 48 47 49 50 34 44 34 49 48 46 50 52 48 46 48 46 48 47 49 50 34 93 44 34 84 121 112 101 34 58 34 79 117 116 66 111 117 110 100 78 65 84 34 125]} {Name:EndpointPolicy Value:[123 34 68 101 115 116 105 110 97 116 105 111 110 80 114 101 102 105 120 34 58 34 49 48 46 48 46 48 46 48 47 49 54 34 44 34 78 101 101 100 69 110 99 97 112 34 58 116 114 117 101 44 34 84 121 112 101 34 58 34 82 79 85 84 69 34 125]}]}.
2018/07/11 00:13:11 [cni-net] Found network azure with subnet 10.240.0.0/12.
2018/07/11 00:13:11 [cni] Calling plugin azure-vnet-ipam ADD nwCfg:&{CNIVersion:0.3.0 Name:azure Type:azure-vnet Mode:bridge Master: Bridge:azure0 LogLevel: LogTarget: Ipam:{Type:azure-vnet-ipam Environment: AddrSpace: Subnet:10.240.0.0/12 Address: QueryInterval:} DNS:{Nameservers:[10.0.0.10 168.63.129.16] Domain: Search:[svc.cluster.local] Options:[]} AdditionalArgs:[{Name:EndpointPolicy Value:[123 34 69 120 99 101 112 116 105 111 110 76 105 115 116 34 58 91 34 49 48 46 50 52 48 46 48 46 48 47 49 50 34 44 34 49 48 46 50 52 48 46 48 46 48 47 49 50 34 93 44 34 84 121 112 101 34 58 34 79 117 116 66 111 117 110 100 78 65 84 34 125]} {Name:EndpointPolicy Value:[123 34 68 101 115 116 105 110 97 116 105 111 110 80 114 101 102 105 120 34 58 34 49 48 46 48 46 48 46 48 47 49 54 34 44 34 78 101 101 100 69 110 99 97 112 34 58 116 114 117 101 44 34 84 121 112 101 34 58 34 82 79 85 84 69 34 125]}]}.
2018/07/11 00:13:11 [cni] Plugin azure-vnet-ipam returned result:<nil>, err:Failed to allocate address: No available addresses.
2018/07/11 00:13:11 [azure-vnet] Failed to allocate address: Failed to delegate: Failed to allocate address: No available addresses.
2018/07/11 00:13:11 [cni-net] ADD command completed with result:Interfaces:[{Name:eth0 Mac: Sandbox:}], DNS:{Nameservers:[] Domain: Search:[] Options:[]} err:<nil>.
2018/07/11 00:13:11 Failed to execute network plugin, err:Failed to allocate address: Failed to delegate: Failed to allocate address: No available addresses.
2018/07/11 00:13:11 Report plugin error
```

kubelet.err.log
```
2018/07/11 00:13:11 "Start Flag true CniSucceeded false Name CNI Version v1.0.7 ErrorMessage Failed to allocate address: Failed to delegate: Failed to allocate address: No available addresses vnet [] 
				Context AzureCNI SubContext "
2018/07/11 00:13:11 OrchestratorDetails &{  kubectl command failed due to exit status 1}
2018/07/11 00:13:11 OSDetails &{windows    }
2018/07/11 00:13:11 SystemDetails &{0 0 0 0 0 0 }
2018/07/11 00:13:11 InterfaceDetails &{Primary 10.240.0.0/12 10.240.0.97 00:0d:3a:06:9e:a1 vEthernet (Ethernet 2) 30 0 }
2018/07/11 00:13:11 BridgeDetails <nil>
2018/07/11 00:13:12 [cni] Timed out on locking store, err:Store is locked.
2018/07/11 00:13:12 Failed to initialize key-value store of network plugin, err:Store is locked.
2018/07/11 00:13:12 Report plugin error
2018/07/11 00:13:12 [cni] Timed out on locking store, err:Store is locked.
2018/07/11 00:13:12 Failed to initialize key-value store of network plugin, err:Store is locked.
2018/07/11 00:13:12 Report plugin error
2018/07/11 00:13:12 Going to send Telemetry report to hostnetagent http://169.254.169.254/machine/plugins?comp=netagent&type=cnireport
2018/07/11 00:13:12 "Start Flag true CniSucceeded false Name CNI Version v1.0.7 ErrorMessage Store is locked vnet [] 
				Context AzureCNI SubContext "
2018/07/11 00:13:12 OrchestratorDetails &{  kubectl command failed due to exit status 1}
2018/07/11 00:13:12 OSDetails &{windows    }
2018/07/11 00:13:12 SystemDetails &{0 0 0 0 0 0 }
2018/07/11 00:13:12 InterfaceDetails &{Primary 10.240.0.0/12 10.240.0.97 00:0d:3a:06:9e:a1 vEthernet (Ethernet 2) 30 0 }
2018/07/11 00:13:12 BridgeDetails <nil>
2018/07/11 00:13:12 Going to send Telemetry report to hostnetagent http://169.254.169.254/machine/plugins?comp=netagent&type=cnireport
2018/07/11 00:13:12 "Start Flag true CniSucceeded false Name CNI Version v1.0.7 ErrorMessage Store is locked vnet [] 
				Context AzureCNI SubContext "
2018/07/11 00:13:12 OrchestratorDetails &{  kubectl command failed due to exit status 1}
2018/07/11 00:13:12 OSDetails &{windows    }
2018/07/11 00:13:12 SystemDetails &{0 0 0 0 0 0 }
2018/07/11 00:13:12 InterfaceDetails &{Primary 10.240.0.0/12 10.240.0.97 00:0d:3a:06:9e:a1 vEthernet (Ethernet 2) 30 0 }
2018/07/11 00:13:12 BridgeDetails <nil>
2018/07/11 00:13:13 SendReport failed due to [Azure CNI] HTTP Post returned statuscode 500
E0711 00:13:13.356110    2804 cni.go:259] Error adding network: netplugin failed but error parsing its diagnostic message "": unexpected end of JSON input
E0711 00:13:13.356110    2804 cni.go:227] Error while adding to cni network: netplugin failed but error parsing its diagnostic message "": unexpected end of JSON input
2018/07/11 00:13:13 [cni] Timed out on locking store, err:Store is locked.
2018/07/11 00:13:13 Failed to initialize key-value store of network plugin, err:Store is locked.
```

**Comment by PatrickLang** [comment]
@sharmasushant is there any mitigation for the IP address exhaustion available?

cc @dineshgovindasamy @dascott @madhanrm

**Comment by SteveCurran** [comment]
@PatrickLang see #3153 same issue since June 4th. Still occurring on 19.1

**Comment by SteveCurran** [comment]
@sharmasushant Unfortunately, that is not the problem. When setting the maxpods each VM interface in Azure gets the same number of secondary ipconfigs as the maxpods. The error arises before reaching the max.

**Comment by sharmasushant** [comment]
@SteveCurran Are you sure about that? Pods in the kube-system name space, such as kube-dns, tiller-deploy, heapster, metrics-server, kubernetes-dashboard, etc. also get ip-addresses from the ones that you have configured.

**Comment by SteveCurran** [comment]
Yes, I am sure. The master node gets its own VM interface with 50 ip addresses. There should be no reason for ip exhaustion.

**Comment by econopsteam** [comment]
Is there any news on this issue? I'm being blocked by the same problem.

**Comment by BIGPHATTOBY** [comment]
Is there any update on this issue? Currently also being blocked

**Comment by ansant** [comment]
Any update on this?

**Comment by PatrickLang** [comment]
This fix is available in ACS-Engine now:
https://github.com/Azure/acs-engine/releases/tag/v0.20.9

If you hit a problem on the new version - can you open a new issue with logs?

Thanks!
